### PR TITLE
Ra 1627

### PR DESCRIPTION
**Pull request by ylatuya**

### CODE REVIEW

File: Xamarin.Forms.Platform.GTK/Controls/ImageButton.cs [hunk 1]
```diff
@@ -117,10 +117,9 @@ public void SetImageFromFile(string fileName)
 			}
 		}
 
-		public override void Dispose()
+		public override void Destroy()
 		{
 			base.Dispose();
```
Comment (vicmaeg) [inline Xamarin.Forms.Platform.GTK/Controls/ImageButton.cs]: shall we call base.Destroy?

File: Xamarin.Forms.Platform.GTK/Cells/ViewCell.cs [hunk 2]
```diff
@@ -51,15 +51,15 @@ private double GetHeightMeasure(Xamarin.Forms.ViewCell viewCell, Gdk.Rectangle a
 			return request.Request.Height;
 		}
 
-		public override void Dispose()
+		public override void Destroy()
```
Comment (vicmaeg) [inline Xamarin.Forms.Platform.GTK/Cells/ViewCell.cs:54]: this is the same that we have in our design guideline, i mean the Dispose is never called and the method that is actually called is the Destroy, is this a known bug in gtk-sharp?

File: Xamarin.Forms.Platform.GTK/Controls/NavigationChildPage.cs [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 namespace Xamarin.Forms.Platform.GTK.Controls
 {
-	public class NavigationChildPage : Gtk.Object
+	public class NavigationChildPage : IDisposable
```
Comment (vicmaeg) [inline Xamarin.Forms.Platform.GTK/Controls/NavigationChildPage.cs:5]: shouldn't this need to be a Gtk.Object?

File: Xamarin.Forms.Platform.GTK/Controls/Page.cs [hunk 4]
```diff
@@ -135,15 +140,15 @@ private void BuildPage()
 
 		private void RefreshToolbar(HBox newToolbar)
 		{
-			_headerContainer.RemoveFromContainer(_toolbar);
```
Comment (vicmaeg) [inline Xamarin.Forms.Platform.GTK/Controls/Page.cs:138]: Why we don't need to call RemoveFromContainer in the HeaderContainer? it is part of the _toolbar?

File: Xamarin.Forms.Platform.GTK/Platform.cs [hunk 5]
```diff
@@ -58,8 +58,7 @@ internal static void DisposeModelAndChildrenRenderers(Element view)
 
 			renderer = GetRenderer((VisualElement)view);
 
-			renderer?.Dispose();
-
+			((Widget)renderer).Destroy();
```
Comment (vicmaeg) [inline Xamarin.Forms.Platform.GTK/Platform.cs]: This was checking for null before, GetRenderer can return a null renderer.

**Review by vicmaeg — CHANGES_REQUESTED** [review]

File: Xamarin.Forms.Platform.GTK/Controls/Page.cs (hunk 4 above)
Comment (ylatuya) [reply Xamarin.Forms.Platform.GTK/Controls/Page.cs:138]: Destroy() already remove it from the container

File: Xamarin.Forms.Platform.GTK/Platform.cs (hunk 5 above)
Comment (ylatuya) [reply Xamarin.Forms.Platform.GTK/Platform.cs]: upps

File: Xamarin.Forms.Platform.GTK/Cells/ViewCell.cs (hunk 2 above)
Comment (ylatuya) [reply Xamarin.Forms.Platform.GTK/Cells/ViewCell.cs:54]: It's explained in the coments of the commit:
https://github.com/fluendo/Xamarin.Forms/pull/3/commits/d159eccb7c59873d05254d01cd12b9af44148750

File: Xamarin.Forms.Platform.GTK/Controls/NavigationChildPage.cs (hunk 3 above)
Comment (ylatuya) [reply Xamarin.Forms.Platform.GTK/Controls/NavigationChildPage.cs:5]: There is no need

File: Xamarin.Forms.Platform.GTK/Cells/ViewCell.cs (hunk 2 above)
Comment (vicmaeg) [reply Xamarin.Forms.Platform.GTK/Cells/ViewCell.cs:54]: 👍

**Review by vicmaeg — APPROVED** [review]

**Pull request merged**